src/clr-angular/utils/focus/focus.service.ts

```diff
@@ -50,12 +50,21 @@ export class FocusService {
     this._unlistenFuncs.push(this.renderer.listen(el, 'keydown.enter', () => !this.activateCurrent()));
   }
 
-  moveTo(item: FocusableItem) {
+  moveTo(item: FocusableItem): boolean {
+    /**
+     * Make sure that item is not undefined
+     */
+    if (item === undefined) {
+      return false;
```

Comment:
I'd like @jeeyun to take a look, but I'd think we'd prefer not to have `moveTo` return a value here. We just want to silently not move focus if there is nothing to focus on. Then you would just `return` and I think leave the rest the same.

Reply:
I agree with you @bdryanovski that our current setup is pretty brittle. Thank you for adding the guard to make sure that the item is defined inside the `moveTo` function.

Even though I suggested earlier that we not introduce the boolean return value, I can see the value in it now. It's up to you if you want to make it return boolean and use that inside the `move` function as you did earlier in this PR. Sorry, I'm not trying to make you go back/forth! 😛 After reading your thought I'm agreeing that returning the boolean value 1. doesn't introduce breaking changes, 2. may add value in its usage. We cannot yet make the function private as you suggested because it's being used in one place outside of the class (in DropdownFocusHandler class).

On a side note, some of the keyboard navigation logic in dropdown seem a bit brittle in general. I'd like for us to review this in the long run to see if we'll keep the current setup or refactor to use the focus directive that we have written since then.

Reply:
I think that you are looking on some older version, cause at my last commit `move()` return only undefined.

And yes, I forgot to submit my comments, I saw it at the demo .. ;)